pkg/memio/memio_ports.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2012-2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package memio
+
+import (
+	"encoding/binary"
+	"io"
+	"os"
+
+	"github.com/u-root/u-root/pkg/ubinary"
+)
+
+type MemIOReader interface {
+	Read(UintN, int64) error
+}
+
+type MemIOWriter interface {
+	Write(UintN, int64) error
+}
+
+type MemIO interface {
+	MemIOReader
+	MemIOWriter
+	io.Closer
+}
+
+type MemIOPort struct {
+	*os.File
+}
+
+func (m *MemIOPort) Read(out UintN, addr int64) error {
+	if _, err := m.File.Seek(addr, io.SeekStart); err != nil {
+		return err
+	}
+	return binary.Read(m.File, ubinary.NativeEndian, out)
+}
+
+func (m *MemIOPort) Write(in UintN, addr int64) error {
+	if _, err := m.File.Seek(addr, io.SeekStart); err != nil {
+		return err
+	}
+	return binary.Write(m.File, ubinary.NativeEndian, in)
+}
+
+func (m *MemIOPort) Close() error {
+	return m.File.Close()
+}
+
```

Comment:
var _ MemIO = &MemIOPort{}

Reply:
I'm pretty sure type assertion looks like this:
`unknownType.(concreteType)`

Do you have a ref doc for your suggested construct of assertion?

Reply:
Ah, never mind.